File: 630-course-schedule-iii/630-course-schedule-iii.cpp
class Solution {
public:
    
    static bool sortbysec(vector<int>& a,
              vector<int>&b)
{
    return (a[1] < b[1]);
}
    
    int scheduleCourse(vector<vector<int>>& v) {
        sort(v.begin(),v.end(), sortbysec);
        long long sum = 0;
       priority_queue<int> q;
     for(auto i : v) {
            sum += i[0];
            q.push(i[0]);
            if(sum > i[1]) {
                sum -= q.top();
                q.pop();
            }
        }
        return q.size();
    }
};

File: 630-course-schedule-iii/630-course-schedule-iii.java
class Solution {
    public int scheduleCourse(int[][] arr) {
         	Arrays.sort(arr, new Comparator<int[]>(){
		@Override
		public int compare(int[] arr1, int[] arr2){
			return arr1[1]-arr2[1];
		}
	});
	PriorityQueue<Integer> q=new PriorityQueue<>(new Comparator<Integer>(){
		@Override
		public int compare(Integer i1, Integer i2){
			return i2-i1;
		}
	});
        long sum = 0;
        for(int i=0;i<arr.length;i++){
            sum+=arr[i][0];
            q.add(arr[i][0]);
            if(sum>arr[i][1]) {
                sum-=q.element();
                q.remove();
            }
        }
        return q.size();

    }
}